logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -162,33 +163,45 @@ public void open() throws IOException {
         for (int pageNum = headCheckpoint.getFirstUnackedPageNum(); pageNum < headCheckpoint.getPageNum(); pageNum++) {
```

Comment:
well ... the head checkpoint is the only way to retrieve the `firstUnackedPageNum` so we have no choice to rely on it. I guess the worst case would be that `firstUnackedPageNum` points to a page that is now fully acked which should not be a problem (need to verify) but if that page has been purged it might be problematic. I will open a followup issue to verify this edge case.